src/relayclient/KnownRelaysManager.ts

```diff
@@ -27,39 +33,55 @@ const activeManagerEvents = ['RelayServerRegistered', 'TransactionRelayed', 'Can
 
 export interface IKnownRelaysManager {
   refresh (): Promise<void>
+
   saveRelayFailure (lastErrorTime: number, relayManager: Address, relayUrl: string): void
-  getRelaysSortedForTransaction (gsnTransactionDetails: GsnTransactionDetails): Promise<RelayRegisteredEventInfo[]>
+
+  getRelaysSortedForTransaction (gsnTransactionDetails: GsnTransactionDetails): Promise<RelayInfoUrl[][]>
+
+  getRelayInfoForManagers (relayManagers: Set<Address>): Promise<RelayRegisteredEventInfo[]>
 }
 
 export default class KnownRelaysManager implements IKnownRelaysManager {
-  private relayFailures = new Map<string, RelayFailureInfo[]>()
-  private readonly activeRelays = new Set<RelayRegisteredEventInfo>()
   private readonly contractInteractor: ContractInteractor
   private readonly config: GSNConfig
   private readonly relayFilter: RelayFilter
   private readonly scoreCalculator: AsyncScoreCalculator
+  private readonly httpClient: HttpClient
 
   private latestScannedBlock: number = 0
+  private relayFailures = new Map<string, RelayFailureInfo[]>()
+
+  public readonly knownRelays: RelayInfoUrl[][] = []
 
-  constructor (contractInteractor: ContractInteractor, config: GSNConfig, relayFilter?: RelayFilter, scoreCalculator?: AsyncScoreCalculator) {
+  constructor (contractInteractor: ContractInteractor, httpClient: HttpClient, config: GSNConfig, relayFilter?: RelayFilter, scoreCalculator?: AsyncScoreCalculator) {
     this.config = config
+    this.httpClient = httpClient
```

Comment:
You don't use it. remove the field
(also it shouldn't use it: for "separation of concerns" , the KnownRelaysManager cares only about the static (on-chain) info, not pigns)

Reply:
fixed